amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -257,4 +266,15 @@ public ShutdownReason getReason() {
         return reason;
     }
 
+    private void dropLease() {
+        Lease currentLease = leaseCoordinator.getCurrentlyHeldLease(leaseKeyProvider.apply(shardInfo));
+        if (currentLease == null) {
+            log.warn("Shard " + shardInfo.shardId() + ": Lease already dropped. Will shutdown the shardConsumer directly.");
+            return;
+        }
```

Comment:
you can move the below if to else condition here

Reply:
Addressed